library/agent/ServiceConfig.ts

```diff
@@ -126,6 +126,35 @@ export class ServiceConfig {
     return { blocked: false };
   }
 
+  private setOnlyAllowedIPAddresses(ipAddresses: IPList[]) {
+    this.onlyAllowedIPAddresses = undefined;
+
+    if (ipAddresses.length === 0) {
+      return;
+    }
+
+    const ips = ipAddresses.map((source) => source.ips).flat();
+
+    this.onlyAllowedIPAddresses = new IPMatcher(ips);
+  }
+
+  updateOnlyAllowedIPAddresses(ipAddresses: IPList[]) {
+    this.setOnlyAllowedIPAddresses(ipAddresses);
+  }
+
+  /**
+   * Returns true if only some IP addresses are allowed to access the service, e.g. if a geoip country allowlist is set.
+   */
+  shouldOnlyAllowSomeIPAddresses() {
+    return this.onlyAllowedIPAddresses !== undefined;
```

Comment:
what if list is empty? Maybe add property on `IPMatcher`?

Reply:
If the ip array is empty, no IPMatcher is created, so it's undefined (also on updates)